test/packagekit_test.dart

```diff
@@ -2073,4 +2166,64 @@ void main() {
         ]));
     await transaction.upgradeSystem('impish', PackageKitDistroUpgrade.stable);
   });
+
+  test('get transaction properties', () async {
+    var server = DBusServer();
+    addTearDown(() async => await server.close());
+    var clientAddress =
+        await server.listenAddress(DBusAddress.unix(dir: Directory.systemTemp));
+    var packagekit = MockPackageKitServer(clientAddress);
+    var t = await packagekit.addTransaction(
+        role: 3,
+        status: 5,
+        lastPackage: 'hal;0.1.2;i386;fedora',
+        uid: 1000,
+        percentage: 42,
+        allowCancel: true,
+        callerActive: false,
+        elapsedTime: 12345,
+        remainingTime: 54321,
+        speed: 299792458,
+        downloadSizeRemaining: 3000000000,
+        transactionFlags: 7);
+    addTearDown(() async => await packagekit.close());
+    await packagekit.start();
+
+    var client = PackageKitClient(bus: DBusClient(clientAddress));
+    addTearDown(() async => await client.close());
+    await client.connect();
+
+    var transaction = await client.getTransaction(t.path);
+
+    expect(transaction.role, equals(PackageKitRole.getDetails));
+    expect(transaction.status, equals(PackageKitStatus.info));
+    expect(transaction.lastPackage, equals('hal;0.1.2;i386;fedora'));
+    expect(transaction.uid, equals(1000));
+    expect(transaction.percentage, equals(42));
+    expect(transaction.allowCancel, equals(true));
+    expect(transaction.callerActive, equals(false));
+    expect(transaction.elapsedTime, equals(12345));
+    expect(transaction.remainingTime, equals(54321));
+    expect(transaction.speed, equals(299792458));
+    expect(transaction.downloadSizeRemaining, equals(3000000000));
+    expect(
+        transaction.transactionFlags,
+        equals({
+          PackageKitTransactionFlag.onlyTrusted,
+          PackageKitTransactionFlag.simulate,
+          PackageKitTransactionFlag.onlyDownload,
+        }));
+
+    await t.client!.emitSignal(
```

Comment:
This can be replaced with `t.emitPropertiesChanged('org.freedesktop.PackageKit.Transaction', changedProperties: {'Percentage': DBusUint32(10)});`

Reply:
:+1: